core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -130,6 +132,71 @@ object Automation {
         }
         return rank
     }
+    
+    // Ranks a tile for the expansion algorithm of cities
+    internal fun rankTileForExpansion(tile: TileInfo, cityInfo: CityInfo): Int {
+        // https://github.com/Gedemon/Civ5-DLL/blob/aa29e80751f541ae04858b6d2a2c7dcca454201e/CvGameCoreDLL_Expansion1/CvCity.cpp#L10301
+        // Apparently this is not the full calculation. The exact tiles are also
+        // dependent on which tiles are between the chosen tile and the city center
+        // Exact details are not implemented, but can be found in CvAStar.cpp:2119,
+        // function `InfluenceCost()`.
+        // Implementing these will require an additional variable for each terrainType
+        val distance = tile.aerialDistanceTo(cityInfo.getCenterTile())
+
+        // Higher score means tile is less likely to be picked
+        var score = distance * 100
+
+        // Resources are good: less points
+        if (tile.hasViewableResource(cityInfo.civInfo)) {
+            val tileResource = tile.getTileResource()
+            if (tileResource.resourceType == ResourceType.Bonus) {
```

Comment:
Sounds like if resourcetype != Bonus || distance<3

Reply:
Note the +1 for bonus resources within range, otherwise this could be simplified indeed. Merging it now would result in everything having to be checked twice:
```
if (resourceType != Bonus || distance < 3) score += -105
if (resourceType == Bonus && distance < 3) score += 1
```